src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
-                if (result != S_OK)
-                {
-                    continue;
-                }
-                wil::unique_cotaskmem_string guidString;
-                if (!SUCCEEDED_LOG(StringFromCLSID(guid, &guidString)))
-                {
-                    continue;
-                }
-
-                std::wstring uuid = guidString.get();
-
-                switch (zoneSetData.type)
-                {
-                case CustomLayoutType::Grid:
-                {
-                    int j = 5;
-                    GridLayoutInfo zoneSetInfo(GridLayoutInfo::Minimal{ .rows = data[j++], .columns = data[j++] });
-
-                    for (int row = 0; row < zoneSetInfo.rows(); row++, j += 2)
-                    {
-                        zoneSetInfo.rowsPercents()[row] = data[j] * 256 + data[j + 1];
-                    }
-
-                    for (int col = 0; col < zoneSetInfo.columns(); col++, j += 2)
-                    {
-                        zoneSetInfo.columnsPercents()[col] = data[j] * 256 + data[j + 1];
-                    }
-
-                    for (int row = 0; row < zoneSetInfo.rows(); row++)
-                    {
-                        for (int col = 0; col < zoneSetInfo.columns(); col++)
-                        {
-                            zoneSetInfo.cellChildMap()[row][col] = data[j++];
-                        }
-                    }
-                    zoneSetData.info = zoneSetInfo;
-                    break;
-                }
-                case CustomLayoutType::Canvas:
-                {
-                    CanvasLayoutInfo info;
-
-                    int j = 5;
-                    info.lastWorkAreaWidth = data[j] * 256 + data[j + 1];
-                    j += 2;
-                    info.lastWorkAreaHeight = data[j] * 256 + data[j + 1];
-                    j += 2;
-
-                    int count = data[j++];
-                    info.zones.reserve(count);
-                    while (count-- > 0)
-                    {
-                        int x = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int y = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int width = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        int height = data[j] * 256 + data[j + 1];
-                        j += 2;
-                        info.zones.push_back(CanvasLayoutInfo::Rect{
-                            x, y, width, height });
-                    }
-                    zoneSetData.info = info;
-                    break;
-                }
-                default:
-                    continue;
-                }
-                customZoneSetsMap[uuid] = zoneSetData;
+            CustomZoneSetJSON result;
 
-                valueLength = ARRAYSIZE(value);
-                dataSize = ARRAYSIZE(data);
+            result.uuid = customZoneSet.GetNamedString(L"uuid");
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.